src/SingleSelectOption.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import { colors, spacers } from './theme.js'
+
+const SingleSelectOption = ({
+    label,
+    active,
+    disabled,
+    onClick,
+    className,
+}) => (
+    <a
```

Comment:
Isn't this more like a button than an anchor?

Reply:
Yeah I guess it is, in the sense that it doesn't trigger navigation. A button's styles will be harder to override though. Would you prefer a button here? I'm all for semantic correctness, but I do wonder if we won't run into a lot of browser defaults making this not look like a button whilst being a button. We could test it, what do you think?

Reply:
I think a button would be the only correct semantic choice here, so I'd go for either `span` or `button`

Reply:
The semantic choices in this context (which is a list): `option`, and `li`.

Those require a different structure altogether so that's why they have been set aside. I do not think that a `button` is a good analogue in this case. I do not think that `a` is a good analogue either, especially as we are forcing it to act like a block element. It does not make sense to me to use a `span` and force it to display as a block, either.

So, since we cannot use the semantic options, and the other two options we have discussed are not semantic _either_, let's go with the neutral version: [The good old workhorse, the `div`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div):

>As a "pure" container, the `<div>` element does not inherently represent anything. **Instead, it's used to group content so it can be easily styled using the class or id attributes**, marking a section of a document as being written in a different language (using the lang attribute), **and so on**.

Emphasis mine. 

Since this is a "fake" input component, i.e. it is not backed by a native form control, we need to think about [keyboard accessibility](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets) and [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) to embed semantics into our custom component.

As there is no semantic choice available, I argue to either keep it as an anchor, or use a semantically neutral element (the `div`).

Reply:
Ok, since we're also using a `div` for the multiselect, I've changed it to a div for the single select option as well: 354dd3aecc121e376f6868da2ff1ecf4727a0fa9

> Since this is a "fake" input component, i.e. it is not backed by a native form control, we need to think about keyboard accessibility and ARIA to embed semantics into our custom component.

Yeah that's a big disadvantage of this component. We lose certain keyboard shortcuts, and for now ARIA semantics also won't be great.

We could add full keyboard shortcuts, but that'll involve attaching refs to all options (at least I see no other option at the moment), which might not be worth it. But I'm not sure, we should probably take a look at that in a separate PR, and properly test what the impact would be.

As for ARIA, implementing that fully would be nice. But I'd have to dig into that. It seems to me this is a good starting point: https://www.w3.org/TR/wai-aria-1.1/#listbox

Reply:
Will add the ARIA todo to the todo list up top.